src/main/java/com/google/api/codegen/viewmodel/DynamicLangXSampleView.java

```diff
@@ -0,0 +1,56 @@
+/* Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.viewmodel;
+
+import com.google.api.codegen.SnippetSetRunner;
+import com.google.auto.value.AutoValue;
+
+/** A ViewModel for standalone samples for dynamic languages. */
+@AutoValue
+public abstract class DynamicLangXSampleView implements ViewModel {
+
+  public abstract String templateFileName();
+
+  @Override
+  public String resourceRoot() {
+    return SnippetSetRunner.SNIPPET_RESOURCE_ROOT;
+  }
+
+  public abstract String outputPath();
+
+  public abstract String className();
+
+  public abstract OptionalArrayMethodView sampledMethod();
```

Comment:
should this be `sampleMethod`?

Reply:
No. With "sampled", I meant to convey that this is the client library method to which this sample pertains, not the interesting method of the sample. Added a comment to clarify

Reply:
Hmm, the wording feels very weird - I've never heard the verb "sample" used in this way, e.g. "this code snippet samples API method X." It sounds like a statistical thing. I would suggest something else like `subjectMethod` or the like.

Reply:
I don't feel strongly about this, so OK, I'll change it to "libraryMethod" and keep the comment I added.

(While I often tend to be a stickler for correct usage, this doesn't bother me much. We neologize all the time in English. The meaning seems clear to me, though granted, I have author's bias)